spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -48,6 +48,31 @@ public int getQueueSize() {
 create a huge number of metrics, because the downstream time series databases
 that need to store the metrics may not deal well with this amount of data.
 
+
+[[enable-disable-config]]
+=== Disabling metrics during deployment
+
+There may be situations where a metric instrumented in an application, or provided by the runtime, are not desired during runtime. MicroProfile Metrics supports the MicroProfile Config property `mp.metrics.enable` to allow users to _disable_ metrics through configuration. Disabled metrics will be blocked from registration and a no-op instance of the metric is returned. Disabled metrics instrumented through annotations will do nothing.
+This property is set at the server level and is read upon server startup. The property follows the following rules:
+
+* The property accepts a semi-colon separated set of values that consists of a metric name followed by a equals sign (`=`) and a `true` or `false` boolean value.
+* The asterisk (`*`) can be used as a wildcard at the end when defining the metric name.
+* Values defined later takes precedence over values before (i.e., right to left precedence).
+
+[[enable-disable-sample]]
+.MicroProfile Config `mp.metrics.enable` property examples
+----
+//metrics with a name that match demo.app.* will be disabled, except the demo.app.testCounter metric
+mp.metrics.enable=demo.app.*=false;demo.app.testCounter=true
+----
+
+CAUTION: Before disabling metrics, be aware if your application uses any `getMetric(...)` or `getCounter(...)`/`getHistogram(...)`/`getTimer(...)`/`getGauge(...)` method calls that try to retrieve a specific metric. Due to the _disabled_ metric not being registered this may cause an `IllegalArgumentException`.
```

Comment:
returns null

Reply:
Suggest using `enabled` (with a trailing `d`) in the config key.

We've talked about choosing config key names which describe states of things (`enabled`) rather than actions to perform on things (`enable`).